fedimint-bin-tests/src/main.rs

```diff
@@ -0,0 +1,630 @@
+#![allow(unused)] // remove when done
+
+use std::cell::RefCell;
+use std::collections::HashMap;
+use std::future::Future;
+use std::ops::{Deref, DerefMut};
+use std::path::PathBuf;
+use std::pin::Pin;
+use std::str::FromStr;
+use std::sync::Arc;
+use std::time::{Duration, Instant};
+
+use anyhow::{Context, Result};
+use bitcoin::hashes::hex::ToHex;
+use bitcoincore_rpc::{bitcoin, RpcApi};
+use cln_rpc::ClnRpc;
+use federation::{run_dkg, Federation};
+use fedimint_client::module::gen::{ClientModuleGenRegistry, DynClientModuleGen};
+use fedimint_core::config::load_from_file;
+use fedimint_core::db::Database;
+use fedimint_core::encoding::Encodable;
+use fedimint_core::task::TaskGroup;
+use fedimint_ln_client::LightningClientGen;
+use fedimint_wallet_client::WalletClientGen;
+use mint_client::modules::mint::MintClientGen;
+use mint_client::{module_decode_stubs, UserClient, UserClientConfig};
+use tokio::fs;
+use tokio::sync::{Mutex, MutexGuard, OnceCell};
+use tokio::time::sleep;
+use tonic_lnd::lnrpc::{GetInfoRequest, Invoice};
+use tracing::{error, info};
+
+mod util;
+
+use std::env;
+
+use util::*;
+
+#[derive(Clone)]
+pub struct Bitcoind {
+    client: Arc<bitcoincore_rpc::Client>,
+    process: ProcessHandle,
+}
+
+impl Bitcoind {
+    pub async fn new(processmgr: &ProcessMgr) -> Result<Self> {
+        let btc_dir = env::var("FM_BTC_DIR")?;
+        let bitcoind = cmd!("bitcoind", "-datadir={btc_dir}");
+        let process = processmgr
+            .spawn_daemon("bitcoind", cmd!("bitcoind", "-datadir={btc_dir}"))
+            .await?;
+
+        let url = env::var("FM_TEST_BITCOIND_RPC")?.parse()?;
+        let (host, auth) = fedimint_bitcoind::bitcoincore_rpc::from_url_to_url_auth(&url)?;
+        let client = Arc::new(bitcoincore_rpc::Client::new(&host, auth)?);
+
+        Self::init(&client).await?;
+        Ok(Self { process, client })
+    }
+
+    async fn init(client: &bitcoincore_rpc::Client) -> Result<()> {
+        // create RPC wallet
+        while let Err(e) = client.create_wallet("", None, None, None, None) {
+            if e.to_string().contains("Database already exists") {
+                break;
+            }
+            error!("Failed to create wallet ... retrying {}", e);
```

Comment:
`warn!`? (I read "error logging level" as "critical logging level"), and this doesn't seem critical and probably can happen sporadically.